### PR TITLE
Update DNSSD records for resolve/browse on common vs commission/operational data

### PR DESCRIPTION
**Pull request by andy31415**
#### Problem
Completely separate data in dns resolution makes it hard to consistently handle both browse and operational discovery for #18256 

#### Change overview
Separates out common and discovery-type specific items as separate structures. Update code to reference the new layout

#### Testing
Ran compilation and unit tests locally and CI will validate tests as well.

### CODE REVIEW

File: src/lib/dnssd/Resolver.h [hunk 1]
```diff
@@ -108,53 +156,15 @@ struct DiscoveredNodeData
...
-        mrpRetryIntervalIdle   = NullOptional;
-        mrpRetryIntervalActive = NullOptional;
-        numIPs                 = 0;
-        port                   = 0;
-        for (unsigned i = 0; i < kMaxIPAddresses; ++i)
-        {
-            ipAddress[i] = chip::Inet::IPAddress::Any;
-        }
+        // NOTE that this assumes 0-fill is ok for this class
+        memset(this, 0, sizeof(*this));
```
Comment (msandstedt) [inline src/lib/dnssd/Resolver.h:163]: Would it be better if the constructor was conventionally initializing members, and then Reset() called the destructor + placement new?

Comment (andy31415) [reply src/lib/dnssd/Resolver.h:163]: Was unsure - placement new/destructor feels a bit hackish.

Is there a benefit? I was assuming memset 0 is very fast and this is what the previous code was doing anyway.

File: src/lib/dnssd/Resolver.h [hunk 2]
```diff
@@ -35,67 +35,115 @@
...
+        memset(hostName, 0, sizeof(hostName));
+        mrpRetryIntervalIdle   = NullOptional;
+        mrpRetryIntervalActive = NullOptional;
+        numIPs                 = 0;
+        port                   = 0;
+        supportsTcp            = false;
+        interfaceId            = Inet::InterfaceId::Null();
+        for (unsigned i = 0; i < kMaxIPAddresses; ++i)
+        {
+            ipAddress[i] = chip::Inet::IPAddress::Any;
```
Comment (bzbarsky-apple) [inline src/lib/dnssd/Resolver.h:90]:
```suggestion
        for (auto & addr : ipAddress)
        {
            addr = chip::Inet::IPAddress::Any;
```

File: src/lib/dnssd/Resolver.h [hunk 3]
```diff
@@ -35,67 +35,115 @@
...
+        {
+            ChipLogDetail(Discovery, "\tHostname: %s", hostName);
+        }
+#if CHIP_DETAIL_LOGGING
+        for (unsigned j = 0; j < numIPs; j++)
+        {
+            char buf[Inet::IPAddress::kMaxStringLength];
+            char * ipAddressOut = ipAddress[j].ToString(buf);
+            ChipLogDetail(Discovery, "\tIP Address #%d: %s", j + 1, ipAddressOut);
+            (void) ipAddressOut;
```
Comment (bzbarsky-apple) [inline src/lib/dnssd/Resolver.h:106]: Why this cast to void?

File: src/lib/dnssd/Resolver.h [hunk 4]
```diff
@@ -192,44 +202,48 @@ struct DiscoveredNodeData
...
+struct ResolvedNodeData
+{
+    CommonResolutionData resolutionData;
+    OperationalNodeData operationalData;
+
+    void LogNodeIdResolved() const
+    {
+#if CHIP_PROGRESS_LOGGING
+        // Would be nice to log the interface id, but sorting out how to do so
+        // across our differnet InterfaceId implementations is a pain.
```
Comment (bzbarsky-apple) [inline src/lib/dnssd/Resolver.h:222]:
```suggestion
        // across our different InterfaceId implementations is a pain.
```
(I know, this just got moved...)

**Review by bzbarsky-apple — APPROVED** [review]

File: src/lib/dnssd/Resolver.h (hunk 2 above)
Comment (andy31415) [reply src/lib/dnssd/Resolver.h:90]: Updated, however this was existing code. Somewhat afraid to not only move things around but also add bugs.

File: src/lib/dnssd/Resolver.h (hunk 1 above)
Comment (msandstedt) [reply src/lib/dnssd/Resolver.h]: I think my critique starts with the fact that the constructor was indirectly calling memset, and that there had to be a comment in the code explaining why this is safe under limited conditions.  It just seems that if this is what we mean to do, we should use c++ tools to do it.  This way, if the class becomes non-trivial in the future, things won't break.

**Review by msandstedt — APPROVED** [review]

**Pull request merged**